#! /bin/sh -e

# DP: Fix capifax for the new (old) ABI.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- capifax/capi.c~	2004-06-30 10:52:29.000000000 +0200
+++ capifax/capi.c	2004-10-26 09:08:46.000000000 +0200
@@ -82,7 +82,7 @@
 			 * of the application
 			 */
 			ALERT_REQ (CMSG, Appl_Id, 0, CONNECT_IND_PLCI(CMSG),
-				   NULL, NULL, NULL, NULL, NULL);
+				   NULL, NULL, NULL, NULL);
 			/* inform the user application */
 			SetState(Connection, D_ConnectPending);
 			IncomingCall(Connection, GetCallingPartyNumber (Connection));
